src/SubscribeBlockTracker.test.ts

```diff
@@ -0,0 +1,3221 @@
+          recordCallsToSetTimeout({ numAutomaticCalls: 1 });
+
+          await withSubscribeBlockTracker(
+            {
+              provider: {
+                stubs: [
+                  {
+                    methodName: 'eth_subscribe',
+                    error: 'boom',
+                  },
+                ],
+              },
+            },
+            async ({ blockTracker }) => {
+              const promiseForCaughtError = new Promise<any>((resolve) => {
+                blockTracker[methodToAddListener]('error', resolve);
+              });
+
+              const promiseForLatestBlock = new Promise((resolve) => {
+                blockTracker[methodToAddListener]('latest', resolve);
+              });
+
+              const caughtError = await promiseForCaughtError;
+              expect(caughtError.message).toStrictEqual('boom');
+              await expect(promiseForLatestBlock).toNeverResolve();
+            },
+          );
+        });
+
+        it('should update the current block number', async () => {
+          recordCallsToSetTimeout();
+
+          await withSubscribeBlockTracker(
+            {
+              provider: {
+                stubs: [
+                  {
+                    methodName: 'eth_blockNumber',
+                    response: {
+                      result: '0x0',
+                    },
+                  },
+                ],
+              },
+            },
+            async ({ blockTracker }) => {
+              await new Promise((resolve) => {
+                blockTracker[methodToAddListener]('latest', resolve);
+              });
+              const currentBlockNumber = blockTracker.getCurrentBlock();
+              expect(currentBlockNumber).toStrictEqual('0x0');
+            },
+          );
+        });
+      });
+
+      describe('"sync"', () => {
+        it('should start the block tracker', async () => {
+          recordCallsToSetTimeout();
+
+          await withSubscribeBlockTracker(({ blockTracker }) => {
+            blockTracker[methodToAddListener]('sync', EMPTY_FUNCTION);
+
+            expect(blockTracker.isRunning()).toBe(true);
+          });
+        });
+
+        it('should take a listener that is called soon after being added', async () => {
+          recordCallsToSetTimeout();
+
+          await withSubscribeBlockTracker(
+            {
+              provider: {
+                stubs: [
+                  {
+                    methodName: 'eth_blockNumber',
+                    response: {
+                      result: '0x0',
+                    },
+                  },
+                ],
+              },
+            },
+            async ({ blockTracker }) => {
+              const sync = await new Promise<Sync>((resolve) => {
+                blockTracker.on('sync', resolve);
```

Comment:
Was this meant to use `methodToAddListener`?

```suggestion
                blockTracker[methodToAddListener]('sync', resolve);
```

Reply:
Yes, thank you! Added in [ab8e7b4](https://github.com/MetaMask/eth-block-tracker/pull/103/commits/ab8e7b43f7d2b9dfde9d4afc83bd20b853de9f60).